plugins/database/mysql/mysql.go

```diff
@@ -0,0 +1,187 @@
+package mysql
+
+import (
+	"database/sql"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/vault/builtin/logical/database/dbplugin"
+	"github.com/hashicorp/vault/helper/strutil"
+	"github.com/hashicorp/vault/plugins/helper/database/connutil"
+	"github.com/hashicorp/vault/plugins/helper/database/credsutil"
+	"github.com/hashicorp/vault/plugins/helper/database/dbutil"
+)
+
+const defaultMysqlRevocationStmts = `
+	REVOKE ALL PRIVILEGES, GRANT OPTION FROM '{{name}}'@'%'; 
+	DROP USER '{{name}}'@'%'
+`
+const mySQLTypeName = "mysql"
+
+type MySQL struct {
+	connutil.ConnectionProducer
+	credsutil.CredentialsProducer
+}
+
+// New implements builtinplugins.BuiltinFactory
+func New() (interface{}, error) {
+	connProducer := &connutil.SQLConnectionProducer{}
+	connProducer.Type = mySQLTypeName
+
+	credsProducer := &credsutil.SQLCredentialsProducer{
+		DisplayNameLen: 4,
+		UsernameLen:    16,
```

Comment:
BTW, "how to deal with this" might be to have different database types that are really just different combinations of credential/connection producers...that was some of the ideas behind the original interfaces. But I'm not sure in a plugin sense how easy this is. (It might be *super* easy by just passing values into the plugin and on the backend side having e.g. a "maria" type that launches a mysql plugin with specified changes.)